File: WEEKLYCODE#4/lisa_workbook/C++/lisa_workbook .cpp
#include <iostream>
using namespace std;

int main() {
    int n, k;
    //taking inputs
    cin>>n;
    cin>>k;
    int answer = 0;
    int page = 1;
    
    //checking for each value of chapter
    for(int chapter = 1; chapter <= n; ++chapter) {
        int problems;
        cin>>problems;
        for(int index = 1; index <= problems; ++index) {
            if(index == page)
                answer++;
            if(index == problems || index % k == 0)
                ++page;
        }
    }
    //printing answer
    cout<<answer<<endl;
    return 0;
}

File: WEEKLYCODE#4/lisa_workbook/Python/lisa_workbook .py
ins = [int(j) for j in input().split()]
#taking inputs
n = ins[0]
k = ins[1]
b = [int(j) for j in input().split()]
answer = 0
page = 0
#checking for each value of chapter
for chapter in b:
    page += 1
    for p in range(1, chapter+1):
        if p == page:
            answer += 1
        if p%k == 0 and p != chapter:
            page += 1
#printing answer
print (answer)